### PR TITLE
fix: resolve Hermes restore and GPU regressions

### PR DESCRIPTION
**Pull request by prekshivyas**
## Summary

The main E2E run exposed two deterministic Hermes GPU regressions and a restore bug whose trigger depends on random token content. This change restores native restart-safe startup proof, preserves exact host evidence for managed GPU fallback while its rollback backup exists, and makes every generated cron drain token parse as one option value.

## Changes

- Pass the Hermes cron drain token as `--drain-token=<value>` so a URL-safe token beginning with `-` cannot be parsed as another option.
- Carry managed-bootstrap authority into native GPU fallback inspection by selecting the full activated container ID and confirming that ID remains exactly labeled.
- Keep the default runtime snapshot rule unchanged: label-only callers still reject zero or multiple matching containers.
- Update the native Hermes GPU live proof to require the restart-safe recreation, persistent startup mode, and `nemoclaw-start` Docker command boundary introduced by #9147.
- Add positive and fail-closed regressions for a retained rollback backup, an absent activated ID, and a leading-hyphen drain token.

## Type of Change

- [x] Code change (feature, bug fix, or refactor)
- [ ] Code change with doc updates
- [ ] Doc only (prose changes, no code sample modifications)
- [ ] Doc only (includes code sample changes)

## Quality Gates

- [x] Tests added or updated for changed behavior
- [ ] Existing tests cover changed behavior — justification:
- [ ] Tests not applicable — justification:
- [ ] Docs updated for user-facing behavior changes
- [x] Docs not applicable — justification: These changes restore existing Hermes restore, managed GPU fallback, and restart-safe startup contracts without changing commands, configuration, defaults, or emitted output.
- [x] Sensitive paths changed (security, policy, credentials, preflight, onboarding, inference, runner, sandbox, or messaging)
- [x] Sensitive-path review completed or maintainer-approved waiver recorded — reviewer/approval link/justification: The full activated Docker ID must be 64 lowercase hexadecimal characters and remain in the exactly labeled result; label-only ambiguity still fails closed, and the drain token remains on the existing direct-argv transport.
- [ ] Non-success, skipped, or missing CI check accepted by maintainer — check name, approval link, and follow-up issue:

## Documentation Writer Review

- [x] Documentation writer subagent reviewed the completed changes
- Result: `no-docs-needed`
- Evidence: Head commit `c24326fba` was independently reviewed across all nine changed files. Existing docs already describe retained rollback backups, exact-container fallback evidence, and the Hermes cron restore gate; the fixes restore those contracts.
- Agent: Codex Desktop

## DGX Station Hardware Evidence

- [ ] Tested on DGX Station
- Tested commit:
- Station profile/scenario:
- Result:
- Supporting evidence:

## Verification

- [x] PR description includes a `Signed-off-by:` line and every commit appears as `Verified` in GitHub
- [x] Normal `pre-commit`, `commit-msg`, and `pre-push` hooks passed, or `npm run validate:pr` passed after refreshing `origin/main` when hooks were skipped or unavailable
- [x] Targeted behavior tests pass for the current change set, or tests are marked not applicable above — command/result or justification: Four focused CLI test files passed, 104 tests total; `npm run typecheck:cli` and `npm run test:projects:check` passed. A separate `npm run test:changed` attempt did not reach a terminal summary locally and was stopped after Vitest became idle; it is not used as evidence.
- [ ] Applicable broad gate passed — `npm test` for broad runtime/test-harness changes; `npm run check` for repo-wide validation/coverage changes — command/result:
- [x] Quality Gates section completed with required justifications or waivers
- [x] No secrets, API keys, or credentials committed
- [ ] `npm run docs` builds without warnings (doc changes only)
- [ ] Doc pages follow the [style guide](https://github.com/NVIDIA/NemoClaw/blob/main/docs/CONTRIBUTING.md) (doc changes only)
- [ ] New doc pages include SPDX header and frontmatter (new pages only)

---
Signed-off-by: Prekshi Vyas <prekshiv@nvidia.com>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved sandbox runtime inspection during managed Docker activation and rollback scenarios.
  * Ensured inspection targets the correct activated container when multiple containers are present.
  * Kept drain tokens securely attached to their command-line option, including unusual token values.
  * Improved GPU sandbox startup validation for restart-safe behavior.

* **Tests**
  * Expanded coverage for runtime inspection, container selection, rollback handling, and drain-token edge cases.

### CODE REVIEW

**Review by senthilr-nv — APPROVED** [review]

File: test/e2e/live/hermes-gpu-startup-proof.ts [hunk 1]
```diff
@@ -76,7 +76,12 @@ export async function assertHermesGpuStartupProof({
     expect(installText).not.toContain(
       "Recreating OpenShell Docker sandbox container with NVIDIA GPU access",
     );
-    expect(installText).not.toContain("Docker container mode selected:");
+    expect(installText).toContain(
+      "Recreating OpenShell Docker sandbox container with restart-safe startup",
+    );
+    expect(installText).toContain(
+      "Docker container mode selected: persistent sandbox startup command",
+    );
```
[the remark this answers is not in the document]
Comment (prekshivyas) [reply test/e2e/live/hermes-gpu-startup-proof.ts:84]: Fixed in c24326fba. The live proof now requires the restart-safe `nemoclaw-start` command boundary and rejects `sleep infinity` for native success as well as compatibility routes. The focused 104-test suite and repository project-membership check pass, and the exact new head passed the required documentation writer review.
